src/battle_ai_util.c

```diff
@@ -2080,6 +2080,20 @@ bool32 HasAnyKnownMove(u32 battlerId)
     return FALSE;
 }
 
+bool32 HasMoveThatLowersOwnStats(u32 battlerId)
+{
+    s32 i;
+    u16 *moves = GetMovesArray(battlerId);
+    for (i = 0; i < MAX_MON_MOVES; i++)
+    {
+        if (moves[i] != MOVE_NONE && moves[i] != MOVE_UNAVAILABLE
+            && gMovesInfo[moves[i]].additionalEffects[i].self
+            && IsOwnStatLoweringEffect(gMovesInfo[moves[i]].additionalEffects[i].moveEffect))
```

Comment:
The `additionalEffects` part isn't correct. it needs it's own loop to loop over the effects a move can have

Reply:
Good catch ty!